sssd_test_framework/utils/sssd.py

```diff
@@ -783,3 +784,52 @@ def autofs(self) -> None:
         """
         self.sssd.authselect.select("sssd")
         self.sssd.enable_responder("autofs")
+
+    def proxy(
+        self,
+        proxy: Literal["files", "ldap"] = "files",
+        provider: str | list[str] = "id",
+        proxy_file: str = "system-auth",
+        sever_hostname: str | None = None,
+        domain: str | None = None,
+    ):
+        """
+        :param proxy: ``ldap`` or ``files``
+        :type proxy: str
+        :param provider: SSSD providers (id, auth, chpass, ...)
+        :type provider: str | list[str]
+        :param proxy_file: Config for proxy
+        :type proxy_file: str
+        :param sever_hostname: LDAP (LDAP server hostname)
+        :type sever_hostname: str
+        :param domain: Existing domain name, defaults to None (= default domain)
+        :type domain: str | None, optional
+        """
+        if domain is None:
+            domain = self.sssd.default_domain
+        if domain is None:
+            raise ValueError("No domain specified!")
+
+        proxy_pam_target = proxy_file
```

Comment:
You use proxy_file as filename here, but as file contents later. What is the intention?

Reply:
It will be the proxy configuration for ex:

       """
            auth    required pam_ldap.so
            account required pam_ldap.so
            password required pam_ldap.so
            session required pam_ldap.so
            """,
If could be different in different test

Reply:
Ok, then I suggest to add two parameters:

* proxy_pam_target: str | None = None - if None, then select `system-auth` for `files` and `sssdproxyldap` for `ldap`
* proxy_pam_stack: str | None = None - if None, choose different default for `files` and `ldap`.

Reply:
It depends what you want. PAM authentication can be handled for files proxy by pam_sss if you require fully qualified user names. In that case, customized PAM config can be used - but of course, default should be system-auth.